CrazyCanvas/Include/GUI/ServerInfo.h

```diff
@@ -5,45 +5,70 @@
 #include "Time/API/Timestamp.h"
 #include "Networking/API/IPEndPoint.h"
 
+#include "Containers/THashTable.h"
+#pragma warning( push, 0 )
+#include <rapidjson/document.h>
+#pragma warning( pop )
+
 #include <string>
+#include "Containers/TArray.h"
 
-struct ServerInfo
-{
-	Noesis::Ptr<Noesis::Grid> ServerGrid;
 
-	std::string Name;
-	std::string MapName;
+	struct ServerInfo
+	{
+		Noesis::Ptr<Noesis::Grid> ServerGrid;
 
-	uint8 Players;
-	uint16 Ping;
-	uint64 ServerUID;
+		std::string Name;
+		std::string MapName;
 
-	LambdaEngine::IPEndPoint EndPoint;
-	LambdaEngine::Timestamp LastUpdate;
+		uint8 Players;
+		uint16 Ping;
+		uint64 ServerUID;
 
+		LambdaEngine::IPEndPoint EndPoint;
+		LambdaEngine::Timestamp LastUpdate;
 
-	bool operator==(const ServerInfo& other) const
-	{
-		return Name == other.Name && MapName == other.MapName && Players == other.Players && Ping == other.Ping && EndPoint == other.EndPoint && ServerUID == other.ServerUID;
-	}
 
-	bool operator!=(const ServerInfo& other) const
-	{
-		return Name != other.Name || MapName != other.MapName || Players != other.Players || Ping != other.Ping || EndPoint != other.EndPoint || ServerUID != other.ServerUID;
-	}
+		bool operator==(const ServerInfo& other) const
+		{
+			return Name == other.Name && MapName == other.MapName && Players == other.Players && Ping == other.Ping && EndPoint == other.EndPoint && ServerUID == other.ServerUID;
+		}
 
-	ServerInfo& operator=(const ServerInfo& other)
-	{
-		if (this != &other)
+		bool operator!=(const ServerInfo& other) const
 		{
-			Name = other.Name;
-			MapName = other.MapName;
-			Players = other.Players;
-			Ping = other.Ping;
-			LastUpdate = other.LastUpdate;
-			EndPoint = other.EndPoint;
-			ServerUID = other.ServerUID;
+			return Name != other.Name || MapName != other.MapName || Players != other.Players || Ping != other.Ping || EndPoint != other.EndPoint || ServerUID != other.ServerUID;
 		}
-		return *this;
-	}
-};
\ No newline at end of file
+
+		ServerInfo& operator=(const ServerInfo& other)
+		{
+			if (this != &other)
+			{
+				Name = other.Name;
+				MapName = other.MapName;
+				Players = other.Players;
+				Ping = other.Ping;
+				LastUpdate = other.LastUpdate;
+				EndPoint = other.EndPoint;
+				ServerUID = other.ServerUID;
+			}
+			return *this;
+		}
+	};
+
+namespace LambdaEngine
+{
+
+	class LAMBDA_API SavedServerSystem
+	{
+	public:
+		DECL_STATIC_CLASS(SavedServerSystem);
+
+		static bool LoadIpsFromFile(TArray<String>& ipAddressess);
+		static bool WriteIpsToFile(const char* ip);
+		static void CreateFile();
+		//static bool IsOnline(const String& action);
+
+	private:
+		static rapidjson::Document s_SavedServerDocument;
+	};
+}
```

Comment:
Is there a special reason why the SavedServerSystem is not a system and is using the DLL import and export macro? And is put in the engine namespace?

Reply:
Nope, just a ctrl + c, ctrl+v. Oopsie